File: tests/fixtures/b2t_typescript_closed_state/after/src/jobs.ts
import { JobState, type JobState as JobStateValue } from "./job-state.js";

export { JobState } from "./job-state.js";
export type { JobState as JobStateValue } from "./job-state.js";

export interface Job {
  state: JobStateValue;
}

export function queue(job: Job): void {
  job.state = JobState.QUEUED;
}

export function isRunning(job: Job): boolean {
  return job.state === JobState.RUNNING;
}

export function isDone(job: Job): boolean {
  return JobState.DONE === job.state;
}

export function isQueuedAlias(job: Job): boolean {
  const currentState = (job.state);
  return currentState === JobState.QUEUED;
}

export function ensureQueued(job: Job): void {
  job.state ??= JobState.QUEUED;
}

export function queueBoth(primary: Job, backup: Job): void {
  primary.state = (backup.state = JobState.QUEUED);
}

File: tests/fixtures/b2p_state_reference/before/app/services/jobs.py
from app.models.jobs import Job


def start(job: Job) -> bool:
    job.status = "queued"
    return job.status == "queued"


def can_finish(job: Job) -> bool:
    return job.status in ("queued", "running")


def is_done(job: Job) -> bool:
    return "done" == job.status
